### PR TITLE
fix: (platform) Split menu button as a component

### PR DESCRIPTION
**Pull request by DeepakSap14**
#### Please provide a link to the associated issue.
https://github.com/SAP/fundamental-ngx/issues/2367

#### Please provide a brief summary of this pull request.
It has a primary Button and a menu dropdown button. These are combined as a single component in Split menu Button.

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done

Documentation checklist:
- [x] update `README.md`
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: libs/platform/src/lib/components/split-menu-button/split-menu-button.component.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    primarButtonClick: EventEmitter<MouseEvent | KeyboardEvent | TouchEvent> = new EventEmitter();
+
+    constructor(protected _cd: ChangeDetectorRef) {
+        super(_cd);
+    }
+
+    /**
+     *  Handles split-menu-button button click
+     */
+    public primarButtonClicked(event: MouseEvent | KeyboardEvent | MouseEvent): void {
```
Comment (sKudum) [inline libs/platform/src/lib/components/split-menu-button/split-menu-button.component.ts:41]: Why only 3 events? Are we not targeting for mobile users? there are 6 more event types.

File: libs/platform/src/lib/components/split-menu-button/split-menu-button.component.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    /** The type of the button.
+     * 'Emphasized', 'Ghost', 'standard', 'positive', 'negative', 'transparent'
+     * Leave empty for default.'*/
+    @Input()
+    type: ButtonType;
+
+    /** Event sent when split-menu-button primary button is clicked */
+    @Output()
+    primarButtonClick: EventEmitter<MouseEvent | KeyboardEvent | TouchEvent> = new EventEmitter();
```
Comment (sKudum) [inline libs/platform/src/lib/components/split-menu-button/split-menu-button.component.ts:32]: Why only 3 events? Are we not targeting for mobile users? there are 6 more event types.

File: libs/platform/src/lib/components/split-menu-button/split-menu-button.component.scss [hunk 3]
```diff
@@ -0,0 +1,5 @@
+@import '~fundamental-styles/dist/button-split';
+
+:host {
```
Comment (sKudum) [inline libs/platform/src/lib/components/split-menu-button/split-menu-button.component.scss:3]: Do we need this?button styles are updated as per Fiori3, could you please recheck do we really need it even with latest version of styles

**Comment by DeepakSap14** [comment]
> Is fdp-button is not useful here?

Tried using fdp-button instead of button, but fdp-button styles does not match with split-menu-button on applying class='fd-button-split'. Attaching screenshot.
Screenshot 2020-04-27 at 2 41 56 PM

File: libs/platform/src/lib/components/split-menu-button/split-menu-button.component.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    styleUrls: ['split-menu-button.component.scss']
+})
+export class SplitMenuButtonComponent extends BaseComponent {
+    /** Label for the first Button */
+    @Input()
+    buttonLabel: string;
+
+    /** reference to menu which will be controlled by split button */
+    @Input()
+    menu: string;
```
Comment (KevinOkamoto) [inline libs/platform/src/lib/components/split-menu-button/split-menu-button.component.ts:18]: I believe the type here should be `MenuComponent`.

File: libs/platform/src/lib/components/split-menu-button/split-menu-button.component.html [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        [fdType]="type"
+        [glyph]="icon"
+        [compact]="contentDensity === 'compact'"
+        (click)="primarButtonClicked($event)"
+    >
+        {{ buttonLabel }}
+    </button>
+    <button
+        fd-button
+        [attr.id]="id"
```
Comment (KevinOkamoto) [inline libs/platform/src/lib/components/split-menu-button/split-menu-button.component.html:17]: The secondary button needs a different `id` than the primary. In your code you should create a new variable, say `idSecond`, and then programmatically compose the `idSecond` by taking `id` and appending a string to it, maybe something like, `-secondary`.

File: libs/platform/src/lib/components/split-menu-button/split-menu-button.component.ts [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+    /** The type of the button.
+     * 'Emphasized', 'Ghost', 'standard', 'positive', 'negative', 'transparent'
+     * Leave empty for default.'*/
+    @Input()
+    type: ButtonType;
+
+    /** Event sent when split-menu-button primary button is clicked */
+    @Output()
+    primarButtonClick: EventEmitter<any> = new EventEmitter();
```
Comment (KevinOkamoto) [inline libs/platform/src/lib/components/split-menu-button/split-menu-button.component.ts:32]: Change spelling to `primaryButtonClick`.

**Comment by DeepakSap14** [comment]
@fkolar please review this.

**Comment by fkolar** [comment]
@SAP/fundamental-denoland-team

**Comment by DeepakSap14** [comment]
All reviews implemented. Rebased with master. merging it now.

**Pull request merged**